refinery/tool_manager/models.py

```diff
@@ -673,6 +673,7 @@ class WorkflowTool(Tool):
     REVERSE = "reverse"
     TOOL_ID = "tool_id"
     WORKFLOW_OUTPUTS = "workflow_outputs"
+    invocation = models.TextField(null=True, blank=True)
```

Comment:
Would it be possible to pass around a variable instead?

Reply:
this is a serious pain-point in terms of usage - everytime `_get_galaxy_workflow_invocation` is called by `_get_workflow_step`, this function is called.  `_get_workflow_step` is used quite a few times.  this is the sort of thing that would get fixed in a refactor.  unlike the other fields i added in the `AnalysisNodeConnection` this one is used repeatedly and makes sense as part of the tool as it is a clear property of it.

Reply:
OK then just drop `null=True` - it is not recommended for `TextField` (see Two Scoops book, 6.2.4)

Reply:
ok, yea, jennifer and i talked about that `blank=True` and `null=True` coexistence at length (since it appears all over the code base) and i remember the outcome of the conversation being to wait for your opinion since so much of it went against convention but we had it nonetheless.  good to know :)